symfit/core/fit_results.py

```diff
@@ -17,32 +22,43 @@ class FitResults(object):
     their optimized values. Can be `**` unpacked when evaluating
     :class:`~symfit.core.fit.Model`'s.
     """
-    def __init__(self, model, popt, covariance_matrix, infodict, mesg, ier, minimizer, objective, **gof_qualifiers):
+
+    def __init__(self, model, popt, covariance_matrix, minimizer_output, minimizer, objective):
         """
         :param model: :class:`~symfit.core.fit.Model` that was fit to.
         :param popt: best fit parameters, same ordering as in model.params.
-        :param pcov: covariance matrix.
-        :param infodict: dict with fitting info.
-        :param mesg: Status message.
-        :param ier: Number of iterations.
+        :param covariance_matrix: covariance matrix.
+        :param minimizer_output: Raw output as given by the minimizer. This is
+            assumed to be a :class:`~scipy.optimize.OptimizationResult` instance.
         :param minimizer: Minimizer instance used.
         :param objective: Objective function which was optimized.
-        :param gof_qualifiers: Any remaining keyword arguments should be
-          Goodness of fit (g.o.f.) qualifiers.
         """
-        # Validate the types in rough way
-        self.infodict = infodict
-        self.status_message = mesg
-        self.iterations = ier
+        self.minimizer_output = minimizer_output
         self.model = model
-        self.gof_qualifiers = gof_qualifiers
         self.minimizer = minimizer
         self.objective = objective
 
         self._popt = popt
-        self.params = OrderedDict([(p.name, value) for p, value in zip(self.model.params, popt)])
+        self.params = OrderedDict(
+            [(p.name, value) for p, value in zip(self.model.params, popt)]
+        )
         self.covariance_matrix = covariance_matrix
+        self.gof_qualifiers = self._gof_qualifiers()
+
+    @cached_property
+    def iterations(self):
+        if hasattr(self.minimizer_output, 'nit'):
+            return self.minimizer_output.nit
```

Comment:
`return getattr(self.minimizer_output, 'nit', None)`, and I wouldn't cache it.

Reply:
Or any other property for FitResult, it just adds complexity for little to no gain.